ios/FluentUI/Tooltip/TooltipViewController.swift

```diff
@@ -9,285 +9,191 @@ import UIKit
-            ambientShadow.shadowOpacity = 1
-            ambientShadow.shadowOffset = CGSize(width: shadowInfo.xOne, height: shadowInfo.yOne)
-            ambientShadow.shadowRadius = shadowInfo.blurOne
+    private static var sourcePointInAnchorView: CGPoint {
+        switch TooltipViewController.arrowDirection {
+        case .up:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width / 2, y: TooltipViewController.anchorView.frame.height)
+        case .down:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width / 2, y: 0)
+        case .right:
+            return CGPoint(x: 0, y: TooltipViewController.anchorView.frame.height / 2)
+        case .left:
+            return CGPoint(x: TooltipViewController.anchorView.frame.width, y: TooltipViewController.anchorView.frame.height / 2)
         }
+    }
 
-        if let perimeterShadow = view.layer.sublayers?[0] {
-            perimeterShadow.frame = view.bounds
-            perimeterShadow.shadowPath = UIBezierPath(roundedRect: view.bounds, cornerRadius: backgroundCornerRadius).cgPath
-            perimeterShadow.shadowColor = UIColor(dynamicColor: shadowInfo.colorTwo).cgColor
-            perimeterShadow.shadowOpacity = 1
-            perimeterShadow.shadowOffset = CGSize(width: shadowInfo.xTwo, height: shadowInfo.yTwo)
-            perimeterShadow.shadowRadius = shadowInfo.blurTwo
-        }
+    private static var sourcePointInWindow: CGPoint {
+        return TooltipViewController.anchorView.convert(sourcePointInAnchorView, to: TooltipViewController.window)
     }
 
-    private static func labelSizeThatFits(_ size: CGSize,
-                                          text: String,
-                                          isAccessibilityContentSize: Bool,
-                                          tokenSet: TooltipTokenSet,
-                                          isMessage: Bool) -> CGSize {
-        let boundingWidth = isAccessibilityContentSize ? size.width : min(tokenSet[.maximumWidth].float - (2 * TooltipTokenSet.paddingHorizontal), size.width)
-        return text.preferredSize(for: UIFont.fluent(tokenSet[isMessage ? .messageLabelTextStyle : .titleLabelTextStyle].fontInfo), width: boundingWidth)
+    private static var boundingRect: CGRect {
+        let screenMargin = TooltipTokenSet.screenMargin
+        return window.bounds.inset(by: window.safeAreaInsets).inset(by: UIEdgeInsets(top: screenMargin,
+                                                                                     left: screenMargin,
+                                                                                     bottom: screenMargin,
+                                                                                     right: screenMargin))
     }
 
-    private func transformForArrowImageView() -> CGAffineTransform {
-        switch positionController.arrowDirection {
+    private let tooltipView: TooltipView
+    private static var anchorView: UIView = UIView()
```

Comment:
these don't need to be static variables?

Reply:
by "these," do you mean all of the variables from lines 174-182? how can they be used in static methods if the they aren't static as well?

Reply:
yes, only determine things that has to be static variables. windows or Views seeems unlikely if tooltipviewwcontroller already has them as properties. arrow direction aslo seems to be not static? because we should only have 1 direction that can change based on the position.

Reply:
I'm not sure how to get around these errors without making the variables static: Instance member 'arrowDirection' cannot be used on type 'TooltipViewController'

Should I make everything non-static and then created a shared instance of TooltipViewController instead like we do in Tooltip.cs?